src/analyses/commonPriv.ml

```diff
@@ -125,20 +125,14 @@ struct
 
   module MustLockset = SetDomain.Reverse (Lockset)
 
-  let rec conv_offset = function
-    | `NoOffset -> `NoOffset
-    | `Field (f, o) -> `Field (f, conv_offset o)
-    (* TODO: better indices handling *)
-    | `Index (_, o) -> `Index (IdxDom.top (), conv_offset o)
-
   let current_lockset (ask: Q.ask): Lockset.t =
     (* TODO: remove this global_init workaround *)
     if !AnalysisState.global_initialization then
       Lockset.empty ()
     else
       let ls = ask.f Queries.MustLockset in
       Q.LS.fold (fun (var, offs) acc ->
-          Lockset.add (Lock.from_var_offset (var, conv_offset offs)) acc
+          Lockset.add (Lock.of_mval (var, Lock.Offs.of_exp offs)) acc
```

Comment:
As far as I can see it, this changes the semantics where we suddenly now have integer offsets in here?
Can we maybe postpone such major semantic changes to a new PR?

Reply:
That was the entire point of the previous `(* TODO: better indices handling *)`. Our must-lockset analysis already tracks definite indices, so there's no reason to suddenly drop them. As far as privatizations are concerned, mutex names are treated opaquely, so `m[1]` is as different from `m[2]` as `m1` is from `m2` or `s1.m` from `s2.m`.

I actually find the old behavior more suspect because `current_lockset` would return `{m}` while events like unlocking would be for `m[2]`. That's odd because we use `Lockset.mem` and `Lockset.remove` in these privatizations and there are no semantic checks on whether `m[2]` and `m` are related/subsumed in any way.

This is another place where using `LockDomain` directly for the query could avoid all such conversions altogether.